mifosng-provider/src/main/java/org/mifosplatform/accounting/glaccount/service/GLAccountReadPlatformServiceImpl.java

```diff
@@ -111,8 +111,10 @@ public GLAccountData mapRow(final ResultSet rs, @SuppressWarnings("unused") fina
         String sql = "select " + rm.schema();
         // append SQL statement for fetching account totals
         if (associationParametersData.isRunningBalanceRequired()) {
-            sql = sql + " and gl_j.id in (select t1.id from (SELECT t2.id , max(concat(t2.entry_date,t2.id)) "
-                    + "FROM acc_gl_journal_entry t2 WHERE is_running_balance_calculated = 1 GROUP BY t2.account_id DESC) t1)";
+            sql = sql + " and gl_j.id in (select t1.id from (select t2.account_id, max(t2.id) as id from "
+                    + "(select id, max(entry_date) as entry_date, account_id from acc_gl_journal_entry group by account_id desc) t3 "
+                    + "inner join acc_gl_journal_entry t2 on t2.account_id = t3.account_id and t2.entry_date = t3.entry_date "
+                    + "group by t2.account_id desc) t1)";
         }
```

Comment:
@vishwasbabu I do. I'll add it and make a new pull request. Thanks for spotting.